numpy/f2py/cfuncs.py

```diff
@@ -548,24 +548,31 @@ def errmess(s: str) -> None:
 #error You need to install NumPy version 0.13 or higher. See https://scipy.org/install.html
 #endif
 """
+
+# Defining the correct value to indicate thread-local storage in C without
+# running a compile-time check (which we have no control over in generated
+# code used outside of NumPy) is hard. Therefore we support overriding this
+# via an external define - the f2py-using package can then use the same
+# compile-time checks as we use for `NPY_TLS` when building NumPy.
```

Comment:
Would it make sense to put a link to where these checks are? i.e. https://github.com/numpy/numpy/blob/7c0e2e4224c6feb04a2ac4aa851f49a2c2f6189f/numpy/_core/meson.build#L246-L271

Reply:
Sure. I added a comment, pointing to the implementation of this in a SciPy PR (that'll be a stable reference, an internal file name may change).